docs/reference/search/search-your-data/search-using-query-rules.asciidoc

```diff
@@ -0,0 +1,150 @@
+* `gte` will match when the rule metadata is greater than or equal to the specified value.
+* `always` will always match for all rule queries.
+--
+* The actions to take when the rule matches a query:
++
+--
+* `ids` will pin the specified <<mapping-id-field,`_id`>>s.
+* `docs` will pin the specified documents in the specified indices.
+--
+
+[discrete]
+[[add-query-rules]]
+=== Add query rules
+
+You can add query rules using the <<put-query-ruleset>> call. This adds a ruleset containing one or more query rules that will be applied to queries that match their specified criteria.
+
+The following command will create a query ruleset called `my-ruleset` with two pinned document rules:
+
+* The first rule will generate a <<query-dsl-pinned-query>> pinning the <<mapping-id-field,`_id`>>s `id1` and `id2` when the `user.query` matches the value `pugs`.
+* The second rule will generate a <<query-dsl-pinned-query>> pinning the <<mapping-id-field, `_id`>> of `id3` specifically from the `my-index-000001` index and `id4` from the `my-index-000002` index when the `user.query` matches the value `beagles`.
+
+[source,console]
+----
+PUT /_query_rules/my-ruleset
+{
+  "rules": [
+    {
+      "rule_id": "rule1",
+      "type": "pinned",
+      "criteria": [
+        {
+          "type": "exact",
+          "metadata": "user.query",
+          "values": [ "pugs" ]
+        }
+      ],
+      "actions": {
+        "ids": [
+          "id1",
+          "id2"
+        ]
+      }
+    },
+    {
+      "rule_id": "rule2",
+      "type": "pinned",
+      "criteria": [
+        {
+          "type": "exact",
+          "metadata": "user.query",
+          "values": [ "beagles" ]
+        }
+      ],
+      "actions": {
+        "docs": [
+          {
+            "_index": "my-index-000001",
+            "_id": "id3"
+          },
+          {
+            "_index": "my-index-000002",
+            "_id": "id4"
+          }
+        ]
+      }
+    }
+  ]
+}
+----
+// TEST[skip:TBD]
+
+The API response returns a results of `created` or `updated` depending on whether this was a new or edited ruleset.
+
+[source,console-result]
+----
+{
+  "result": "created"
+}
+----
+// TEST[skip:TBD]
+
+You can use the <<get-query-ruleset>> call to retrieve the ruleset you just created,
+the <<list-query-rulesets>> call to retrieve a summary of all query rulesets,
+and the <<delete-query-ruleset>> call to delete a query ruleset.
+
+[discrete]
+[[rule-query-search]]
+=== Perform a rule query
+
+Once you have defined a query ruleset, you can search this ruleset using the <<query-dsl-rule-query>> query.
+An example query for the `my-ruleset` defined above is:
+
+[source,console]
+----
+GET /my-index-000001/_search
+{
+  "query": {
+    "rule_query": {
+      "organic": {
+        "query_string": {
+          "query": "pugs"
+        }
+      },
+      "match_criteria": {
+        "query_string": "pugs"
+      },
```

Comment:
Would adding more match_criteria fields be confusing? I think it would help users understand the matching power of the query rules:

```suggestion
      "match_criteria": {
        "query_string": "pugs",
        "user_id": "kimchy"
      },
```

Reply:
Added!